MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -1951,42 +2012,52 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
             var videoStream = state.VideoStream;
             var filters = new List<string>();
 
+            var videoDecoder = GetHardwareAcceleratedVideoDecoder(state, options);
+            var inputWidth = videoStream?.Width;
+            var inputHeight = videoStream?.Height;
+            var threeDFormat = state.MediaSource.Video3DFormat;
+
+            // When the input may or may not be hardware VAAPI decodable
+            if (string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase) && options.EnableHardwareEncoding)
```

Comment:
I feel that we need a separate toggle of `allow hw decoding` along with `allow hw encoding`... out of scope of this PR, though.

Reply:
Or we can use the `-all` option in newer version `vainfo` tool to output all hardware accelerated formats supported by the VAAPI device and parse the output into a configuration file. In addition, the toggle of QSV/NVENC will be unchecked after playback fails. I think we need a tool that can detect all hardware decoding format support. In that case, it is also convenient to show the user whether the hardware decoding and encoding are used.

Reply:
> In addition, the toggle of QSV/NVENC will be unchecked after playback fails.

This should no longer occur, it was fixed by @dkanada I think quite a while ago.
What we really need, IMHO, are ways of controlling both HW decode and HW encode separately, **and** a fallback to software decode if HW decode fails for whatever reason.